### ISSUE TITLE
Add support for partitions in NDM os disk filter

### ISSUE THREAD

**Issue by akhilerm** [openebs/openebs#3133]
## Description

Currently the NDM os disk filter excludes the complete disk even if one partition contains the mount point specified in the NDM config map.
eg: Consider the following case. A single disk is available on the host, with multiple partitions. One partition is used for `/` and one for the boot section.
```
sda      8:0    0 465.8G  0 disk 
├─sda1   8:1    0   487M  0 part /boot/efi
├─sda5   8:5    0 114.5G  0 part /
├─sda6   8:6    0 339.4G  0 part
└─sda7   8:7    0  11.5G  0 part
```
With a configmap as follows:
```
filterconfigs:
      - key: os-disk-exclude-filter
        name: os disk exclude filter
        state: true
        exclude: "/,/etc/hosts,/boot"
```
currently the complete disk `/dev/sda` will be exlcuded by NDM, and blockdevice will not be created. But there is an unused partition of 339GB which can be used as a blockdevice. With NDM  support for partitons now available, the support for partitions on OS disk should be there.

If the support is added, there will be blockdevice resource created for the partition `/dev/sda6` and `/dev/sda7` which can be claimed and used by applications.

## Context

There are kubernetes clusters which do not support adding additional disks to the nodes. But the root disk will be of huge capacity. If this feature is added, the users can create partitions on the main disk and openebs will be able to use the extra space available on the disk.

## Possible Solution

The os-disk-exclude filter should be modified such that the mounted device itself is exlcuded instead of the parent device. The changes should be done [here](https://github.com/openebs/node-disk-manager/blob/287edbe59a6a9a48d897799c204822aa79afe29d/cmd/ndm_daemonset/filter/osdiskexcludefilter.go#L91) in the os disk filter.

**Comment by asoltesz** [comment]
I also need this functionality.

I am using Hetzner Cloud machines. I can partition the disk (sda1 for the OS, sda2 for OpenEBS). I cannot have the free space as a separate, standalone device (sdb)

Currently, I cannot utilize sda2 because OpenEBS always ignores it due to os-disk-exclude.

**Comment by akhilerm** [comment]
```
NAME   FSTYPE MOUNTPOINT   SIZE
sda                        1.8T
├─sda1                     500G
├─sda2                     500G
├─sda3                     500G
└─sda4 ext4   /kubernetes  363G
sdb                       55.9G
├─sdb1 vfat   /boot/efi    512M
└─sdb2 ext4   /           55.4G
```

When having the following disk configuration on the nodes(kubernetes volumes on a partition on a different disk other than root disk), NDM is excluding both `sda` and `sdb`, making the partitions on `sda` unusable. NDM should be able to only exclude the partitions specified with the mountpoint instead of the complete disk.

**Comment by OBAA** [comment]
This has not been fixed! I had to abandon openEBS months ago. Decided to try again and realised this was my hold back. Block devices wont create as long as the partition is on the same disks as the mount path in the exclude filter. I had to replace

> exclude: "/,/etc/hosts,/boot"

with
> exclude: " "

and **manually excluded the partitions** mounted in this paths "/,/etc/hosts,/boot" using **path-filter**
